tests/service_configuration/test_service_configuration.py

```diff
@@ -99,3 +99,54 @@ def get_param(name):
         assert conf_file_data["browser_settings"]["CUSTOM"]["window_size"] == "0x0"
 
     os.remove(conf_file_path)
+
+
+@patch("urldownloader.subprocess.run")
+def test_service_conf_default(mock_run):
+
+    service_default_user_agent = "DEFAULT_USER_AGENT"
+    user_custom_window_size = "CUSTOM_WINDOW_SIZE"
+
+    ud = service_class(config={"default_browser_settings": {"user_agent": service_default_user_agent}})
+
+    mock_request = create_autospec(ServiceRequest)
+    mock_request.file_name = "testfile1"
+    mock_request.file_type = "uri/http"
+    mock_request.task = create_autospec(Task)
+
+    mock_request.task.fileinfo = create_autospec(FileInfo)
+    mock_request.task.fileinfo.uri_info = URIInfo(
+        {
+            "uri": "http://test.test.not_real",
+            "scheme": "http",
+            "netloc": "http://test.test.not_real",
+            "hostname": "test.test.not_real",
+        }
+    )
+
+    def get_param(name):
+        params = {"proxy": "no_proxy"}
+        return params.get(name, None)
+
+    mock_request.get_param = get_param
+
+    # Default should be run when no task param set for uri
+    try:
+        ud.execute_kangooroo(mock_request, {}, {"window_size": user_custom_window_size})
+    except Exception:
+        pass
+
+    # check parameter values that gets passed to Kangoorooo
+    run_args = mock_run.call_args.args[0]
+    conf_file_path = run_args[3]
+    browser_setting_type = run_args[8]
+
+    assert browser_setting_type == "CUSTOM"
+
+    with open(conf_file_path, "r") as tmp_conf_file:
+        conf_file_data = yaml.safe_load(tmp_conf_file)
+        assert service_default_user_agent == conf_file_data["browser_settings"]["DEFAULT"]["user_agent"]
+
+        assert user_custom_window_size == conf_file_data["browser_settings"]["CUSTOM"]["window_size"]
```

Comment:
We should add
`assert None == conf_file_data["browser_settings"]["CUSTOM"].get("user_agent")`
To make sure it will rely on the defaut.